redis_hashring/__init__.py

```diff
@@ -320,14 +320,13 @@ def poll(self, as_greenlet=False):
 
         try:
             while True:
+                # Since Redis' listen method blocks, we use select to inspect the
+                # underlying socket to see if there is activity.
+                fileno = pubsub.connection._sock.fileno()
                 timeout = max(0, POLL_INTERVAL - (time.time() - last_heartbeat))
-                message = pubsub.get_message(
-                    ignore_subscribe_messages=True, timeout=timeout
-                )
-                if message:
-                    # Pull remaining messages off of channel.
-                    while pubsub.get_message():
-                        pass
+                r, w, x = self._select([fileno], [], [], timeout)
+                if fileno in r:
+                    next(gen)
```

Comment:
... instead of `next(gen)` we should just
```
                    while pubsub.get_message():
                         pass
```

That way it will consume all of the message, and not just one.

Reply:
Done in https://github.com/closeio/redis-hashring/pull/16/commits/3a1221c1bf833b3ebffe6d2d4352c4de3f6c1cec.